### ISSUE TITLE
feature: accept a context tag name to `Nokogiri::HTML5::DocumentFragment.parse`

### ISSUE THREAD

**Issue by flavorjones** [sparklemotion/nokogiri#3183]
Currently, the XML and HTML4 document fragment parse methods accept a context tag name, but the HTML5 class does not.

This has made implementing code like https://github.com/flavorjones/nokogiri-html5-inference a little bit more challenging. It's a useful API and we should just support it.

This would be a good first issue for someone!

**Comment by sharvy** [comment]
Hi @flavorjones, if no one has started working on this issue, I would love to work on it.

**Comment by flavorjones** [comment]
@sharvy Ah, thank you so much for offering!

I apologize, but I already implemented this feature in #3246 (see eca68e7e), but I neglected to close this issue because I referenced the wrong issue id in that PR.

This feature will be in v1.17 when it ships, hopefully at the end of July.

Are you interested in this feature specifically? Or can I point you at some other issue(s) that might be a good "first issue"?
